### PR TITLE
Implemented retire DiagnosticReport (Encounter)

### PR DESCRIPTION
**Pull request by milankarunarathne**
@surangak @harsha89 Implemented [Retire existing Diagnostic Report](https://issues.openmrs.org/browse/FM-162).
Please, review the code.

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java [hunk 1]
```diff
@@ -150,12 +153,25 @@ public DiagnosticReport saveFHIRDiagnosticReport(DiagnosticReport diagnosticRepo
...
-		// Delete `Status` Obs
-
-		// Delete Obs (`Result` as Set of Obs)
+	public void retireFHIRDiagnosticReport(String id) {
+		EncounterService encounterService = Context.getEncounterService();
 		// Delete Binary Obs Handler which used to store `PresentedForm`
-		return diagnosticReport;
+
+		// Delete Encounter OpenMRS Object
+		Encounter omrsDiagnosticReport = encounterService.getEncounterByUuid(id);
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java]: So we are only need to retire the encounter? Is there any additional resources need to retire as well? May be Obs associate with the encounter retired when encounter retires by underline API?

File: api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java [hunk 2]
```diff
@@ -150,12 +153,25 @@ public DiagnosticReport saveFHIRDiagnosticReport(DiagnosticReport diagnosticRepo
...
+
+		if (omrsDiagnosticReport == null) {
```
Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java]: So we are only need to retire the encounter? Is there any additional resources need to retire as well? May be Obs associate with the encounter retired when encounter retires by underline API?

Comment (surangak) [inline api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java]: @harsha89 , as per the API, I do believe that the underlying Obs are retired when you retire an encounter. At least, that's what happens according to: http://resources.openmrs.org/doc/org/openmrs/api/EncounterService.html#voidEncounter-org.openmrs.Encounter-java.lang.String-

Comment (harsha89) [inline api/src/main/java/org/openmrs/module/fhir/api/diagnosticreport/handler/LaboratoryHandler.java]: Cool merging the pull request.

**Pull request merged**